src/pages/sportDetail.tsx

```diff
@@ -1,83 +1,146 @@
-        types: [],
-        paticipatingCountries: [],
-    },
-    {
-        sportId: 5,
-        name: 'Badminton',
-        summary: 'Badminton is a racket sport whose exact origins are still a tantalising mystery. It evolved from the older game of Battledor and Shuttlecock, which was a popular pastime in Europe, particularly among the affluent classes. However, it is not known exactly when Battledore and Shuttlecock transformed into the competitive sport of badminton. One plausible theory is that badminton was first played at the stately home of the Duke of Beaufort in Gloucestershire sometime in the early 1860s and thus was named after his estate, which was called Badminton House. The game travelled to India where it became a popular sport in the military cantonments. Gradually it spread in the British colonies and then to Europe and East Asia. Badminton today is a global sport with widespread appeal for people of all ages and abilities.',
-        types: [],
-        paticipatingCountries: [],
-    },
-    {
-        sportId: 6,
-        name: 'Basketball',
-        summary: 'Basketball was invented by James W. Naismith to keep his students fit during the winter. In December 1891, the physical education teacher at the YMCA International Training School in Springfield, USA sought a suitable indoor sport - and most of his rules still apply today. In the 1920s, the first international games were played and the first World Championships for Men and Women took place in the 1950s.',
-        types: [],
-        paticipatingCountries: [],
-    },
-];
 
+type SportParams = {
+    id: string;
+};
 
 const SportsDetail = () => {
-    const { id } = useParams();
+    const { id } = useParams<SportParams>();
     const navigate = useNavigate();
 
-    const sport = sportsData.find((sport) => sport.sportId === parseInt(id, 10));
+    const [sport, setSport] = useState<Sport | null>();
+    const [loading, setLoading] = useState(true);
+
+    const fetchSport = async (sportId: string) => {
+        setLoading(true);
+        try {
+            const res = await fetch(`https://sota-backend.fly.dev/sport/${sportId}`);
+            const data: Sport = await res.json();
+            if (Object.keys(data).length !== 0) {
+                setSport(data);
+            } else {
+                navigate(`/sports/1`)
+            }
+        } catch (error) {
+            console.error("Error fetching sport data: ", error);
+            navigate(`/sports/1`);
+        } finally {
+            setLoading(false);
+        }
+    };
+
+    useEffect(() => {
+        if (Number(id) > 0 && id) { fetchSport(id); }
```

Comment:
```suggestion
        if (id && parseInt(id, 10) > 0) { fetchSport(id); }
```

Please check for null before parsing integer.